test/e2e/runner/load.go

```diff
@@ -67,58 +65,47 @@ func Load(ctx context.Context, testnet *e2e.Testnet, multiplier int) error {
 }
 
 // loadGenerate generates jobs until the context is canceled
-func loadGenerate(ctx context.Context, chTx chan<- types.Tx, multiplier int) {
-	for i := 0; i < math.MaxInt64; i++ {
-		// We keep generating the same 1000 keys over and over, with different values.
-		// This gives a reasonable load without putting too much data in the app.
-		id := i % 1000
+func loadGenerate(ctx context.Context, txCh chan<- types.Tx, testnet *e2e.Testnet, id []byte) {
+	t := time.NewTimer(0)
+	<-t.C
+	for {
+		for i := 0; i < testnet.LoadTxBatchSize; i++ {
+			tx, err := payload.NewBytes(&payload.Payload{
+				Id:          id,
+				Size:        uint64(testnet.LoadTxSizeBytes),
+				Rate:        uint64(testnet.LoadTxBatchSize),
+				Connections: uint64(testnet.LoadTxConnections),
+			})
+			if err != nil {
+				panic(fmt.Sprintf("Failed to generate tx: %v", err))
+			}
 
-		bz := make([]byte, 1024) // 1kb hex-encoded
-		_, err := rand.Read(bz)
-		if err != nil {
-			panic(fmt.Sprintf("Failed to read random bytes: %v", err))
-		}
-		tx := types.Tx(fmt.Sprintf("load-%X=%x", id, bz))
+			select {
+			case txCh <- tx:
 
+			case <-ctx.Done():
+				close(txCh)
+				return
+			}
+		}
+		t.Reset(time.Second)
 		select {
-		case chTx <- tx:
-			time.Sleep(time.Second / time.Duration(multiplier))
-
+		case <-t.C:
```

Comment:
Do we want here a mechanism similar to [this one](https://github.com/informalsystems/tm-load-test/blob/a517a0ac52f5c26e0b1f6a480ff871ee70edba03/pkg/loadtest/transactor.go#L274) in `tm-load-test`, to avoid the possibility that the batches pile up?

Reply:
Added.